### PR TITLE
UI tests for restricting the sharing

### PR DESCRIPTION
**Pull request by individual-it**
## Description
UI test to see if various ways of restricting the sharing options work e.g "Share with group membership only" #29241

This PR contains also the changes from #29388 because it depends on them

## Related Issue
#29075

## Motivation and Context
test sharing behaviour 

## How Has This Been Tested?
run tests locally, travis will test again

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by individual-it** [comment]
@jvillafanez could you please check if the tests described in https://github.com/owncloud/core/pull/29406/commits/f0a4b4d8205f9bcf2d34f0c8d495ff8d8c5f807c#diff-4a8972ffadc126987f05eebd52151aec are what you have programmed
And if you can up with more sensible checks, that would be great!

File: tests/ui/features/sharing/restrictReSharing.feature [hunk 1]
```diff
@@ -0,0 +1,40 @@
+Feature: restrict resharing
+
+As admin
+I would like to be able to forbid the sharing of a received share globaly
```
Comment (phil-davis) [inline tests/ui/features/sharing/restrictReSharing.feature]: s/globaly/globally/

**Review by phil-davis — COMMENTED** [review]
Just little things.
This seems a good start to testing the sharing restrictions. After this we can easily look through the other sharing settings and make basic UI tests to confirm that other combinations work.

**Comment by PVince81** [comment]
I suggest rebasing for CI

**Comment by jvillafanez** [comment]
I think we should include tests for the case where the checkbox aren't checked. Other than that, the tests look fine.

**Comment by individual-it** [comment]
added one more test and rebased again

**Review by phil-davis — APPROVED** [review]
LGTM - assuming CI passes

**Comment by individual-it** [comment]
backport in #29470

**Pull request merged**